src/main/java/org/edumips64/ui/swing/DineroFrontend.java

```diff
@@ -500,15 +515,29 @@ public DineroCacheOptions(char type, int level) {
 	}
 	
 	public String toString() {
-		String prefix = "-l" + level + "-" + type;
-		String cmdline = prefix + "size" + " " + size + " ";
-		cmdline += prefix + "bsize" + " " + bsize + " ";
-
-		if(assoc > 0)
-			cmdline += prefix + "assoc" + " " + assoc + " ";
-		if(ccc)
-			cmdline += prefix + "ccc" + " ";
+		// creates empty builder, capacity 16
+		StringBuilder cacheConfig = new StringBuilder();
+		// adds 9 character string at beginning
+		try {
+			String prefix = "-l" + level + "-" + type;
+			cacheConfig.append("prefix");
```

Comment:
This will just append to the stringbuilder the string `"prefix"`, not the variable `prefix`.

Also, `String.format` is probably a better way to create the options instead of concatenation.

So for example the prefix can be created and appended this way:

```
String prefix = String.Format("-l%d-%s", level, type);

cacheConfig.append(prefix);
```

To this, you can then create the other parts of the command line and append them (size, bsize, etc). No need to add them to cmdline any more.

For example, for size:

```
cacheConfig.append(String.Format("size %s", size));
```

Does this make sense?

Reply:
I have done the string builder part much, but I will update once I finish it. I recall it was a previous change that you requested. But for now, I think I understand what you have mentioned here.

Reply:
Would it be alright if I did not use StringBuilder at all? Because the use of the string is very limited and using an object may be less beneficial than just a String variable.